sdk/eventhub/event-hubs/src/dataTransformer.ts

```diff
@@ -86,6 +87,14 @@ export const defaultDataTransformer = {
       if (isRheaAmqpSection(body)) {
         switch (body.typecode) {
           case dataSectionTypeCode:
+            if (isBuffer(body.content) && body.content.byteLength === 0) {
+              // byte array of length zero indicates empty body
+              return {
+                body: null,
+                bodyType: "data",
+              };
+            }
```

Comment:
So, if an empty buffer was sent, a buffer won't be received?

Reply:
Right, based on the discussion in the internal messaging crew channel, using empty byte array to indicate empty data section seems a convention in amqp space.  This could potentially break customers though.  Since there's no direct customer feedback, we could table it now until later when we bump major version?